src/main/java/com/ldtteam/structurize/commands/ScanCommand.java

```diff
@@ -62,77 +64,98 @@
+
+    private static int onExecuteWithPlayerNameAndFileName(final CommandContext<CommandSource> context) throws CommandSyntaxException
+    {
+        final BlockPos from = BlockPosArgument.getBlockPos(context, POS1);
+        final BlockPos to = BlockPosArgument.getBlockPos(context, POS2);
+        GameProfile profile = GameProfileArgument.getGameProfiles(context, PLAYER_NAME).stream().findFirst().orElse(null);
+        String name = StringArgumentType.getString(context, FILE_NAME);
+        return execute(context.getSource(), from, to, Optional.empty(), profile, name);
+    }
+
+    private static int onExecuteWithPlayerNameAndFileNameAndAnchorPos(final CommandContext<CommandSource> context) throws CommandSyntaxException
+    {
+        final BlockPos from = BlockPosArgument.getBlockPos(context, POS1);
+        final BlockPos to = BlockPosArgument.getBlockPos(context, POS2);
+        final BlockPos anchorPos = BlockPosArgument.getBlockPos(context, ANCHOR_POS);
+        GameProfile profile = GameProfileArgument.getGameProfiles(context, PLAYER_NAME).stream().findFirst().orElse(null);
+        String name = StringArgumentType.getString(context, FILE_NAME);
+        return execute(context.getSource(), from, to, Optional.of(anchorPos), profile, name);
+    }
+
     protected static LiteralArgumentBuilder<CommandSource> build()
     {
         return newLiteral(NAME)
-                 .then(newArgument(POS1, BlockPosArgument.blockPos())
-                         .then(newArgument(POS2, BlockPosArgument.blockPos())
-                                 .executes(ScanCommand::onExecute)
-                                 .then(newArgument(PLAYER_NAME, GameProfileArgument.gameProfile())
-                                         .executes(ScanCommand::onExecute)
-                                         .then(newArgument(FILE_NAME, StringArgumentType.word())
-                                                 .executes(ScanCommand::onExecute)))));
+                .then(newArgument(POS1, BlockPosArgument.blockPos())
+                        .then(newArgument(POS2, BlockPosArgument.blockPos())
+                                .executes(ScanCommand::onExecute)
+                                .then(newArgument(ANCHOR_POS, BlockPosArgument.blockPos())
+                                        .executes(ScanCommand::onExecuteWithAnchor))
+                                .then(newArgument(PLAYER_NAME, GameProfileArgument.gameProfile())
+                                        .executes(ScanCommand::onExecuteWithPlayerName)
+                                        .then(newArgument(FILE_NAME, StringArgumentType.word())
+                                                .executes(ScanCommand::onExecuteWithPlayerNameAndFileName)
+                                                .then(newArgument(ANCHOR_POS, BlockPosArgument.blockPos())
+                                                        .executes(ScanCommand::onExecuteWithPlayerNameAndFileNameAndAnchorPos))))));
```

Comment:
really way much duplicate code, look at vanilla how to reduce it

Reply:
vanilla doesn't have any commands structured like this

Reply:
almost every multiargument command has that reduction :P

Reply:
eg. you can create lambda with context and pass booleans with using ternary operator for getting

Reply:
This is fine.
It is clear and lambdas woulds not shrink the overload, this is at least clearly readable which is the best you can hope for.